### PR TITLE
Popup Disappearing problem solved

### LINKED ISSUE #407: Popup not disappearing
the tool-tip for inserting table and some other tool-tips do not disappear on clicking somewhere else on screen and only disappear only when clicked on a specific button like in case of table we need to click the table button again to hide the tool-tip.
Screenshot from 2020-02-24 03-00-10
I have seen similar issue like this in plots2 repo but not sure if they are related to this or not

### PR DESCRIPTION
**Pull request by NitinBhasneria**
Fixes: https://github.com/publiclab/PublicLab.Editor/issues/407

Presently the toolbar popup is not disappearing. After this fix, if we click anywhere else the popup will disappear.
popup
In this PR, a new crossvent function is added, this function will be called when we click anywhere except the popup box.

### CODE REVIEW

**Comment by NitinBhasneria** [comment]
cc: @jywarren @emilyashley @VladimirMikulic @shreyaa-s @keshav234156 @Shulammite-Aso.

File: src/prompts/prompt.js [hunk 1]
```diff
@@ -58,6 +59,14 @@ function prompt (options, done) {
   function focusDialog () {
     dom.input.focus();
   }
+  
+  function rootClick (e) {
+    var str = e.target.classList.value;
+    console.log(str);
```
Comment (VladimirMikulic) [inline src/prompts/prompt.js]: console logs shouldn't be present in the source code.

File: src/prompts/prompt.js [hunk 2]
```diff
@@ -58,6 +59,14 @@ function prompt (options, done) {
...
+    if (!str.includes('wk-prompt')){
```
Comment (VladimirMikulic) [inline src/prompts/prompt.js]: )_{ (_ -> space)

**Review by VladimirMikulic — CHANGES_REQUESTED** [review]
Please resolve minor issues.

**Comment by NitinBhasneria** [comment]
We can also add a scroll event. By this when we scroll the page the popup will disappear, which in turn makes the user experience better.
@VladimirMikulic @shreyaa-s @keshav234156 @Shulammite-Aso @jywarren . What do you think?

**Comment by Shulammite-Aso** [comment]
Yes @NitinBhasneria I think anything  that means focusing outside of the pop-up should make it disappear.

**Comment by VladimirMikulic** [comment]
@NitinBhasneria maybe it's not a bad idea but it should trigger on a "bigger" scroll. Triggering popup removal on a slight scroll (a few pixels) which I sometimes accidentally do (others probably do as well) on my trackpad would be annoying to the users.

File: src/prompts/prompt.js [hunk 3]
```diff
@@ -58,6 +59,13 @@ function prompt (options, done) {
   function focusDialog () {
     dom.input.focus();
   }
+  
+  function rootClick (e) {
+    var str = e.target.classList.value;
```
Comment (VladimirMikulic) [inline src/prompts/prompt.js]: Please rename `str` to something meaningful. Comment above the function would also be appreciated.

**Review by VladimirMikulic — CHANGES_REQUESTED** [review]
One last change :)

**Comment by NitinBhasneria** [comment]
Ok thanks for the idea @jywarren @VladimirMikulic

**Comment by NitinBhasneria** [comment]
@VladimirMikulic @jywarren  Have a look now. 
I have added onscroll function which will make the popup disappear after more than 10 scrolls.
popup

**Comment by jywarren** [comment]
This looks really good! but @VladimirMikulic were you saying you think it may need to be on a scroll of more than X pixels, rather than a repeated scroll? 

Just to clarify before we get ready to merge this!

And, once we decide the exact behavior we want, @NitinBhasneria can you rename the title and first line of description above to clearly lay out the change in behavior? This will help us produce good documentation and track all the changes that go into each release.

Thank you!!! 🙌

File: src/prompts/prompt.js [hunk 4]
```diff
@@ -35,12 +36,21 @@ function prompt (options, done) {
...
   crossvent.add(dom.close, 'click', remove);
   crossvent.add(dom.ok, 'click', ok);
   crossvent.add(dom.input, 'keypress', enter);
   crossvent.add(dom.dialog, 'keydown', esc);
   classify(dom, options.classes.prompts);
 
+  // Disappearing the popup when scrolled.
+  window.onscroll= function() {
+    scrollSize++;
+    if (scrollSize>10) {
```
Comment (VladimirMikulic) [inline src/prompts/prompt.js]: This should be formatted correctly.

File: src/prompts/prompt.js [hunk 5]
```diff
@@ -58,6 +68,13 @@ function prompt (options, done) {
   function focusDialog () {
     dom.input.focus();
   }
+  
+  function rootClick (e) {
+    var classlist = e.target.classList.value;
```
Comment (VladimirMikulic) [inline src/prompts/prompt.js]: `classlist` is too generic. Which element is this that we are referencing? Also, follow the camelCase naming convention.

File: src/prompts/prompt.js [hunk 6]
```diff
@@ -58,6 +68,13 @@ function prompt (options, done) {
   function focusDialog () {
     dom.input.focus();
   }
+  
+  function rootClick (e) {
```
Comment (VladimirMikulic) [inline src/prompts/prompt.js:73]: The function needs to be documented.

**Review by VladimirMikulic — CHANGES_REQUESTED** [review]

**Comment by VladimirMikulic** [comment]
@jywarren Nitin covered the case I mentioned. :+1:

**Comment by jywarren** [comment]
OK, great! Thanks, everyone! @NitinBhasneria once this is done could you open a PR to bump the version number on `woofmark` so that we can publish it and pull it into the editor repo? Thanks!!

**Comment by NitinBhasneria** [comment]
> OK, great! Thanks, everyone! @NitinBhasneria once this is done could you open a PR to bump the version number on `woofmark` so that we can publish it and pull it into the editor repo? Thanks!!

created #6

**Pull request merged**